src/text-editor.coffee

```diff
@@ -101,8 +101,8 @@ class TextEditor extends Model
     throw new Error("Must pass a config parameter when constructing TextEditors") unless @config?
     throw new Error("Must pass a clipboard parameter when constructing TextEditors") unless @clipboard?
     throw new Error("Must pass a grammarRegistry parameter when constructing TextEditors") unless @grammarRegistry?
-    throw new Error("Must pass an assert parameter when constructing TextEditors") unless @assert?
 
+    @assert ?= @defaultAssert.bind(this)
```

Comment:
I think I am fine with this way of handling it: we could even use a global, but I like that all the dependencies get listed in this object's constructor.

I find `defaultAssert` to be somewhat confusing, though. How would you feel about inlining it to show it's a no-op?

```coffee
@assert ?= (condition) => condition
```

And if we find ourselves repeating this pattern over and over, we could extract a file like `no-op-assert.js' and `require` it:

```coffee
@assert ?= require('no-op-assert')
```

What do you think?